src/natcap/invest/ui_server.py

```diff
@@ -102,6 +104,60 @@ def get_invest_validate():
     return json.dumps(results)
 
 
+@app.route('/colnames', methods=['POST'])
+def get_vector_colnames():
+    """Get a list of column names from a vector.
+    This is used to fill in dropdown menu options in a couple models.
+
+    Body (JSON string):
+        vector_path (string): path to a vector file
+        
+    Returns:
+        a JSON string.
+    """
+    payload = request.get_json()
+    LOGGER.debug(payload)
+    vector_path = payload['vector_path']
+    colnames = []  
+    # a lot of times the path will be empty so don't even try to open it
+    if vector_path:  
+        try:
+            vector = gdal.OpenEx(vector_path, gdal.OF_VECTOR)
+            colnames = [defn.GetName() for defn in vector.GetLayer().schema]
+        except:
+            LOGGER.error(f'Could not read column names from {vector_path}')
```

Comment:
I don't mind catching the broad exception here because I think flask would catch it anyway, but is it worth logging the actual exception here as well?